### PR TITLE
SNOW-197585 aws glue options

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
SNOW-197585 and #412 
Fixes an issue where on AWS Glue `setuptools` doesn't know what packages are installed.

### CODE REVIEW

File: src/snowflake/connector/options.py [hunk 1]
```diff
@@ -12,14 +13,15 @@
...
 
 def warn_incompatible_dep(dep_name: str,
                           installed_ver: str,
                           expected_ver: 'pkg_resources.Requirement') -> None:
     warnings.warn(
         "You have an incompatible version of '{}' installed, please install a version that "
-        "adheres to: '{}'".format(dep_name,
-                                  _expected_version),
+        "adheres to: '{}'".format(dep_name, _expected_pyarrow_version),
         stacklevel=2)
```
Comment (sfc-gh-wshangguan) [inline src/snowflake/connector/options.py:25]: Do we want to use function parameters like `expected_ver` instead of `_expected_pyarrow_version`?

Comment (sfc-gh-mkeller) [reply src/snowflake/connector/options.py:25]: Oops, right. My bad!

File: src/snowflake/connector/options.py [hunk 2]
```diff
@@ -2,53 +2,103 @@
...
+    """
+    try:
+        pandas = importlib.import_module('pandas')  # NOQA
+        # since we enable relative imports without dots this import gives us an issues when ran from test directory
+        from pandas import DataFrame  # NOQA
+        import pyarrow  # NOQA
+        pyarrow = importlib.import_module('pyarrow')  # NOQA
+        # Check whether we have the currently supported pyarrow installed
+        installed_packages = pkg_resources.working_set.by_key
+        print(installed_packages)
```
Comment (sfc-gh-stakeda) [inline src/snowflake/connector/options.py:74]: Is this intentional?

Comment (sfc-gh-stakeda) [reply src/snowflake/connector/options.py:74]: or a debug message?

Comment (sfc-gh-mkeller) [reply src/snowflake/connector/options.py:74]: Oops, I'll look over the PR one more time. Sorry!

**Comment by sfc-gh-mkeller** [comment]
> In general, it would be better splitting a refactor change and actual change?

The header change I could split into its own PR, but the options refactor was prompted by me not being able to test this behavior before.

File: src/snowflake/connector/options.py [hunk 3]
```diff
@@ -2,53 +2,102 @@
...
-    keyring = None
+    If available it returns pandas and pyarrow packages with a flag of whether they were imported.
+    It also warns users if they have an unsupported pyarrow version installed if possible.
+    """
+    try:
+        pandas = importlib.import_module('pandas')  # NOQA
+        # since we enable relative imports without dots this import gives us an issues when ran from test directory
+        from pandas import DataFrame  # NOQA
+        import pyarrow  # NOQA
+        pyarrow = importlib.import_module('pyarrow')  # NOQA
```
Comment (sfc-gh-wshangguan) [inline src/snowflake/connector/options.py:71]: May I know why do we need both of these two lines?

Comment (sfc-gh-mkeller) [reply src/snowflake/connector/options.py:71]: I forgot to remove the old import line, sorry!

**Comment by sfc-gh-mkeller** [comment]
Note: I have moved the copyright header to this PR: #420 
I also want to apologize for not looking over this more carefully beforehand!

**Review by sfc-gh-kwagner — APPROVED** [review]
LGTM

**Review by sfc-gh-wshangguan — APPROVED** [review]
LGTM!

**Review by sfc-gh-stakeda — APPROVED** [review]

**Pull request merged**